bagitobjecttransfer/recordtransfer/views.py

```diff
@@ -953,97 +953,129 @@ def get_in_progress_submission(user: User, uuid: str) -> Optional[InProgressSubm
 
 
 @require_http_methods(["POST"])
-def upload_file(request: HttpRequest) -> JsonResponse:
-    """Upload a single file to the server, and return a token representing the file upload
-    session. If a token is passed in the request header using the Upload-Session-Token header, the
-    uploaded file will be added to the corresponding session.
+def create_upload_session(request: HttpRequest) -> JsonResponse:
+    """Create a new upload session and return the session token.
+
+    Args:
+        request: The POST request sent by the user.
+
+    Returns:
+        JsonResponse: The session token of the newly created upload session.
+    """
+    try:
+        user: User = cast(User, request.user)
+        session = UploadSession.new_session(user=user)
+        return JsonResponse({"uploadSessionToken": session.token}, status=201)
+    except Exception as exc:
+        LOGGER.error("Error creating upload session: %s", str(exc), exc_info=exc)
+        return JsonResponse(
+            {"error": gettext("There was an internal server error. Please try again.")},
+            status=500,
+        )
+
+
+@require_http_methods(["GET", "POST"])
+def upload_or_list_files(request: HttpRequest, session_token: str) -> JsonResponse:
+    """Upload a single file to the server list the files uploaded in a given upload session. The
+    file is added to the upload session using the session token passed as a parameter in the
+    request. If an session token is invalid, an error message is returned.
```

Comment:
One tiny change needed in this docstring:

```suggestion
    request. If a session token is invalid, an error message is returned.
```